src/microbe_stage/PlayerMicrobeInput.cs

```diff
@@ -297,6 +297,40 @@ public void CloseSignalingCommandsMenu()
             stage.HUD.ApplySignalCommand(command, stage.Player);
     }
 
+    [RunOnKeyDown("g_sprint")]
+    public bool StartSprint()
+    {
+        if (!stage.HasPlayer || !stage.GameWorld.WorldSettings.ExperimentalFeatures)
+            return true;
+
+        ref var control = ref stage.Player.Get<MicrobeControl>();
+        control.Sprinting = true;
+
+        // We need to not consume the input, otherwise the key up for this will not run
+        return false;
+    }
+
+    [RunOnKeyUp("g_sprint")]
+    public void EndSprint()
+    {
+        if (!stage.HasPlayer || !stage.GameWorld.WorldSettings.ExperimentalFeatures)
```

Comment:
I don't think this needs to check experimental features, sprinting should always be able to end even if experimental features are not on.